### PR TITLE
Changing maxtasksperchild to default 

### PR DESCRIPTION
**Pull request by girifb**
## WHY

While running with hydra submitit plugin on slurm, we noticed that `_try_multiprocess` method deadlocks in `pool.terminate()` call, causing train step to hang.

## REPRODUCING STEPS

One can reproduce this by setting up a hydra config: `tests/hydra/config/hydra_npg_config.yaml` and a python script `tests/hydra/hydra_policy_opt_job_script.py`

and run these commands:

```
python hydra_policy_opt_job_script.py --multirun hydra/launcher=submitit_local
python hydra_policy_opt_job_script.py --multirun hydra/launcher=submitit_slurm
```

### CODE REVIEW

File: tests/hydra/config/hydra_npg_config.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# Algorithm hyperparameters : if alg requires additional params, can be specified here (or defaults will be used)
+
+alg_hyper_params  :   {}
+
+hydra:
+          launcher:
+                      cpus_per_task: 12
+                      gpus_per_node: 0
+                      tasks_per_node: 1
```
Comment (vikashplus) [inline tests/hydra/config/hydra_npg_config.yaml:36]: A  few of these options don't comply with local launchers. We might want to fix this before merging.

File: tests/hydra/hydra_policy_opt_job_script.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
+"""
+This is a job script for running policy gradient algorithms on gym tasks.
```
Comment (vikashplus) [inline tests/hydra/hydra_policy_opt_job_script.py:2]: This job_script is a duplicate of the original job script with a few changes to support hydra. We should think if we want to default to hydra for everything or we should keep this as separate hydra_specific script. @aravindr93?

Comment (aravindr93) [reply tests/hydra/hydra_policy_opt_job_script.py:2]: I think it will be good to move to hydra for everything, but lets fix all the corner cases before a major refactor.

File: tests/hydra/config/hydra_npg_config.yaml (hunk 1 above)
Comment (aravindr93) [reply tests/hydra/config/hydra_npg_config.yaml:36]: Since this is just a test script, we can leave it at this for now. Once we move everything to hydra, we will have a clean default config that is ideal for our machine types. If cpus_per_task is more than available, I think hydra defaults to max available.

**Review by aravindr93 — APPROVED** [review]
I think this fixes the multiprocessing issue with the master branch, so I'll accept the PR now. The problem still persists with the v2 branch.
